src/create.cpp

```diff
@@ -331,25 +369,58 @@ namespace create {
     return serial->send(cmd, 5);
   }
 
-  bool Create::driveWheels(const float& leftVel, const float& rightVel) const {
-    int16_t leftCmd = roundf(leftVel * 1000);
-    int16_t rightCmd = roundf(rightVel * 1000);
-    BOUND(leftCmd, -util::CREATE_2_MAX_VEL * 1000, util::CREATE_2_MAX_VEL * 1000);
-    BOUND(rightCmd, -util::CREATE_2_MAX_VEL * 1000, util::CREATE_2_MAX_VEL * 1000);
+  bool Create::driveWheels(const float& leftVel, const float& rightVel) {
+    float boundedLeftVel = BOUND_CONST(leftVel, -model.maxVelocity, model.maxVelocity);
+    float boundedRightVel = BOUND_CONST(rightVel, -model.maxVelocity, model.maxVelocity);
+    requestedLeftVel = boundedLeftVel;
+    requestedRightVel = boundedRightVel;
+    if (model.version > V_1) {
+      int16_t leftCmd = roundf(boundedLeftVel * 1000);
+      int16_t rightCmd = roundf(boundedRightVel * 1000);
+
+      uint8_t cmd[5] = { OC_DRIVE_DIRECT,
+                         rightCmd >> 8,
+                         rightCmd & 0xff,
+                         leftCmd >> 8,
+                         leftCmd & 0xff
+                       };
+      return serial->send(cmd, 5);
+    } else {
+      float radius;
+      // Prevent divide by zero when driving straight
+      if (boundedLeftVel != boundedRightVel) {
+        radius = -((model.axleLength / 2.0) * (boundedLeftVel + boundedRightVel)) /
+          (boundedLeftVel - boundedRightVel);
+      } else {
+        radius = util::STRAIGHT_RADIUS;
+      }
+
+      float vel;
+      // Fix signs for spin in place
+      if (boundedLeftVel == -boundedRightVel || std::abs(roundf(radius * 1000)) <= 1) {
```

Comment:
Again, take care when comparing floating points this way. I think you want  `if (boundedLeftVel - boundedRightVel < EPS)`